### PR TITLE
[primTorch] Remove prims.to_dtype

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92380

cc @ezyang @mruberry @ngimel @Lezcano

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -4428,7 +4426,7 @@ def logspace(
         pin_memory=pin_memory,
         requires_grad=requires_grad,
     )
-    return prims.to_dtype(torch.pow(base, ret), dtype)
+    return _maybe_convert_to_dtype(torch.pow(base, ret), dtype)
```
Comment (lezcano) [inline torch/_refs/__init__.py:4429]: Even better, you can pass `dtype=dtype` to `linspace` and it should work as expected.

**Review by lezcano — APPROVED** [review]
Cool

File: torch/_refs/__init__.py (hunk 1 above)
Comment (peterbell10) [reply torch/_refs/__init__.py:4429]: I would presume the `pow` calculation is intentionally done in float64 for precision reasons.

**Review by ngimel — COMMENTED** [review]
Do you want to remove `to_dtype` itself from _prims/__init__.py ?

**Review by ngimel — APPROVED** [review]

File: torch/_refs/__init__.py (hunk 1 above)
Comment (lezcano) [reply torch/_refs/__init__.py:4429]: aaaand, I miss the pow. Sorry for that.

**Pull request closed without merging**